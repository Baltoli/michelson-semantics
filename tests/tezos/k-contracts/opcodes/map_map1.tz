$contract { parameter nat;
storage (map string nat);


code { UNPAIR; SWAP;
       MAP { CDR; DIP {DUP}; ADD; };
       DIP { DROP; };
       NIL operation; PAIR; }
 } ;
$param 10 ;
$storage { Elt "foo" 1 } ;
$ExpectedOutput { Elt "foo" 11 } ;

