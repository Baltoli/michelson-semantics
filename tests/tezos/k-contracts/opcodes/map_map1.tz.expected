<generatedTop>
  <k>
    .
  </k>
  <stack>
    Pair .List #Map ( "foo" |-> 11 , string .AnnotationList , nat .AnnotationList )
  </stack>
  <paramtype>
    nat .AnnotationList
  </paramtype>
  <storagetype>
    map .AnnotationList string .AnnotationList nat .AnnotationList
  </storagetype>
  <balance>
    #Mutez ( 0 )
  </balance>
  <amount>
    #Mutez ( 0 )
  </amount>
  <now>
    #Timestamp ( -1 )
  </now>
  <myaddr>
    #Address ( "InvalidMyAddr" )
  </myaddr>
  <knownaddrs>
    .Map
  </knownaddrs>
  <sourceaddr>
    #Address ( "InvalidSourceAddr" )
  </sourceaddr>
  <senderaddr>
    #Address ( "InvalidSenderAddr" )
  </senderaddr>
  <chainid>
    #ChainId ( 0 )
  </chainid>
</generatedTop>
